cardano-crypto-tests/src/Test/Crypto/DSIGN.hs

```diff
@@ -31,166 +136,159 @@ import Test.Tasty.QuickCheck (testProperty)
+-- corresponding verification key.
+prop_dsign_verify  
+  :: forall (v :: Type) (a :: Type) . 
+  (DSIGNAlgorithm v, ContextDSIGN v ~ (), Signable v a) 
+  => (a, SignKeyDSIGN v)
   -> Property
-prop_dsign_verify_pos a sk =
-  let sig = signDSIGN () a sk
+prop_dsign_verify (msg, sk) = 
+  let signed = signDSIGN () msg sk
       vk = deriveVerKeyDSIGN sk
-  in verifyDSIGN () vk a sig === Right ()
+    in verifyDSIGN () vk msg signed === Right ()
 
--- | If we sign a message @a@ with one signing key, if we try to verify the
--- signature (and message @a@) using a verification key corresponding to a
--- different signing key, then the verification fails.
---
-prop_dsign_verify_neg_key
-  :: forall v. (DSIGNAlgorithm v, Eq (SignKeyDSIGN v),
-                ContextDSIGN v ~ (), Signable v ~ SignableRepresentation)
-  => Message
-  -> SignKeyDSIGN v
-  -> SignKeyDSIGN v
+-- If we sign a message with one key, and try to verify with another, then
+-- verification fails.
+prop_dsign_verify_wrong_key
+  :: forall (v :: Type) (a :: Type) .
+  (DSIGNAlgorithm v, Signable v a, ContextDSIGN v ~ ()) 
+  => (a, SignKeyDSIGN v, SignKeyDSIGN v) 
   -> Property
-prop_dsign_verify_neg_key a sk sk' =
-  sk /= sk' ==>
-    let sig = signDSIGN () a sk
-        vk' = deriveVerKeyDSIGN sk'
-    in verifyDSIGN () vk' a sig =/= Right ()
-
+prop_dsign_verify_wrong_key (msg, sk, sk') = 
+  let signed = signDSIGN () msg sk
+      vk' = deriveVerKeyDSIGN sk'
+    in verifyDSIGN () vk' msg signed =/= Right ()
 
--- | If we sign a message @a@ with one signing key, if we try to verify the
--- signature with a message other than @a@, then the verification fails.
---
-prop_dsign_verify_neg_msg
-  :: forall v. (DSIGNAlgorithm v,
-                ContextDSIGN v ~ (), Signable v ~ SignableRepresentation)
-  => Message
-  -> Message
-  -> SignKeyDSIGN v
+-- If we signa a message with a key, but then try to verify with a different
+-- message, then verification fails.
+prop_dsign_verify_wrong_msg 
+  :: forall (v :: Type) (a :: Type) .
+  (DSIGNAlgorithm v, Signable v a, ContextDSIGN v ~ ())
+  => (a, a, SignKeyDSIGN v)
   -> Property
-prop_dsign_verify_neg_msg a a' sk =
-  a /= a' ==>
-    let sig = signDSIGN () a sk
-        vk = deriveVerKeyDSIGN sk
-    in verifyDSIGN () vk a' sig =/= Right ()
-
---
--- Arbitrary instances
---
-
-instance DSIGNAlgorithm v => Arbitrary (VerKeyDSIGN v) where
-  arbitrary = deriveVerKeyDSIGN <$> arbitrary
-  shrink = const []
-
-instance DSIGNAlgorithm v => Arbitrary (SignKeyDSIGN v) where
-  arbitrary = genKeyDSIGN <$> arbitrarySeedOfSize seedSize
-    where
-      seedSize = seedSizeDSIGN (Proxy :: Proxy v)
-  shrink = const []
-
-instance (DSIGNAlgorithm v,
-          ContextDSIGN v ~ (), Signable v ~ SignableRepresentation)
-      => Arbitrary (SigDSIGN v) where
-  arbitrary = do
-    a <- arbitrary :: Gen Message
-    sk <- arbitrary
-    return $ signDSIGN () a sk
-  shrink = const []
+prop_dsign_verify_wrong_msg (msg, msg', sk) = 
```

Comment:
As with `prop_dsign_verify_wrong_key`, the `/=` precondition matters; it probably matters even more here, because messages are subject to more aggressive shrinking (or at least, they can be). E.g., if messages are `String`s, then the shrinker will come up with the test case where both messages are the empty string.

Reply:
Same as before; see `genWrongMsg`.